x-pack/legacy/plugins/ml/public/services/new_job_capabilities_service.ts

```diff
@@ -199,4 +204,14 @@ function addEventRateField(aggs: Aggregation[], fields: Field[]) {
   fields.splice(0, 0, eventRateField);
 }
 
+// remove fields which are text and have a keyword equivalent
+function processTextFields(fields: Field[]) {
```

Comment:
It's worth noting here that for the categorization field choice we actually prefer `text`, but for by/over/partition/influencer fields, it's best to use `keyword`. If too difficult to get this in for this PR, maybe one we can think about when we add a categorization job wizard?

Reply:
agreed. i'll follow this PR up with one that adjusts the new job capabilities service to produce  a special list of fields for the categorization field list